internal/authctx/client.go

```diff
@@ -38,26 +38,28 @@ func (cfg *TanzuContext) Setup() (err error) {
 	if err != nil {
 		return err
 	}
+	cfg.TMCConnection.WithHost(cfg.ServerEndpoint)
```

Comment:
these changes are not needed, can you please explain why this change needed ?

Reply:
Same as explained in below, trying to move out setting host in headers to where `setup()` is called, so that we can reuse `setup()` for refreshing auth tokens